daylight/pmap/README.md

```diff
@@ -1,2 +1,13 @@
 # photon mapping
-A folder that includes resources related to radiance photon mapping!
+
+A folder that includes resources related to Radiance photon mapping!
+
+
+## Links of interest
+
+* Wikipedia: [Photon Mapping]( https://en.wikipedia.org/wiki/Photon_mapping (
```

Comment:
@theo-armour the closing parentheses is backwards!

I assigned reviewing the definition to @rschregle who really knows the topic. It looks right to me but he is the expert to confirm this.

Reply:
Hi and sorry for taking so long to reply to this. I've read your synopsis, which describes photon mapping in general, and just thought I'd add some comments specific to the RADIANCE implementation:
* Photons are only emitted from the light sources (via _mkpmap_), as such it's strictly a forward raytracer.
* Subsurface scattering isn't directly supported, though I imagine such a case could be (tediously) constructed with _mist_.
* Photon mapping is essentially a locally biased variant of bidirectional path tracing, with RADIANCE's backward raytracer completing the paths towards the viewer.
* Photons carry flux in Watts or Lumen; the irradiance resp. illuminance is proportional to the local photon density.
* The local bias results from the nearest neighbour lookup to evaluate the photon density; since the neighbouring photons occupy a finite surface area, this implies a local energy transfer which manifests itself as visual blurring.
* Bias can be reduced by reducing the lookup bandwidth parameter for _rpict/rtrace_, etc., but this increases noise. This bias/noise tradeoff is a fundamental problem of density estimation algorithms in general.
* There's a bias compensation algorithm available by specifying a bandwidth range (lower/upper bound). This was published in a [paper](https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1467-8659.2003.00720.x) that systematically analysed the various factors contributing to bias, and proposed a statistical model to detect it, and adjust the bandwidth accordingly. While it works, it's slow and no consistency is enforced for neighbouring pixels, so the resulting noise from bias reduction is quite apparent. Numerically, this yields pretty accurate (if unstable, for low sample counts) results. More that anything, this was an academic exercise and raised awareness of the fact that photon mapping is far from the perfect rendering algorithm.
* With the exception of volume photons, photons are always surface-bound; consequently density estimates should also be surface-bound to reduce bias, although this isn't enforced. However, photons are not actually stored **on** the surfaces, but are decoupled from them in 3D space. You can actually render a photon map with geometry other than the one it was built with; again, nothing is enforced here; user beware.
* Photon paths are statistically independent and thus emitted in parallel on multicore platforms via _mkpmap_'s -n option (currently not supported on Windoze).

I might add that the RADIANCE photon mapping implementation is seriously outdated -- kinda like the rest of RADIANCE. ;^) 
Nowadays photons are no longer used to directly evaluate illuminance/irradiance, but rather to guide more sophisticated sampling algorithms by providing a coarse approximation of illumination to construct probability density functions for paths. Vertex Connection and Merging (VCM) is one such example, which combines photon mapping with sophisticated path tracing. Its quality and performance is light-years from traditional photon mapping.

Current photon mappers are also not static like the RADIANCE implementation; instead of running to completion, they employ progressive techniques that iteratively accumulate renderings from many small photon maps to approximate a very large photon map, while providing visual updates between iterations. At the same time, the lookup bandwidth is progressively reduced to reduce bias, while accumulating (averaging) the renderings reduces noise.
I dabbled a bit in a progressive frontend for the RADIANCE photon map in the context of daylight redirection, which was published in a [paper](https://www.sciencedirect.com/science/article/pii/S0038092X15000559). I've since ported the code to Python, but it remains experimental and Greg refuses to include any more Python code in the official RADIANCE distribution. The code is available as-is on request tho.

Hope this helps. Let me know if I should edit the definition directly.

Reply:
Thank you @rschregle! @theo-armour where should we go from here?

Also, @rschregle the Python implementation sounds like something that fits pretty well in what we are doing/trying to do. At this point I'm trying not to be distracted from the core functionalities but @sariths might want to give it a try.